src/components/SelectableMainTable.tsx

```diff
@@ -20,6 +20,9 @@ interface SelectableMainTableProps {
   setSelectedNames: (val: string[]) => void;
   processingNames: string[];
   rows: MainTableRow[];
+  // NOTE: indeterminate names are only effective during initialisation of this component
+  indeterminateNames?: Set<string>;
+  unSelectableNames?: Set<string>;
```

Comment:
Maybe instead of introducing this unSelectableNames concept, we can give those rows an empty name, so they become disabled implicitly?